slack_sdk/socket_mode/aiohttp/__init__.py

```diff
@@ -0,0 +1,193 @@
+
+from slack_sdk.socket_mode.async_client import AsyncBaseSocketModeClient
+from slack_sdk.socket_mode.async_listeners import (
+    AsyncWebSocketMessageListener,
+    AsyncSocketModeRequestListener,
+)
+from slack_sdk.socket_mode.request import SocketModeRequest
+from slack_sdk.web.async_client import AsyncWebClient
+
+
+class SocketModeClient(AsyncBaseSocketModeClient):
+    logger: Logger
+    web_client: AsyncWebClient
+    app_token: str
+    wss_uri: Optional[str]
+    auto_reconnect_enabled: bool
+    web_socket_message_queue: Queue
+    web_socket_message_listeners: List[
+        Union[
+            AsyncWebSocketMessageListener,
+            Callable[
+                ["AsyncBaseSocketModeClient", dict, Optional[str]], Awaitable[None]
+            ],
+        ]
+    ]
+    socket_mode_request_listeners: List[
+        Union[
+            AsyncSocketModeRequestListener,
+            Callable[["AsyncBaseSocketModeClient", SocketModeRequest], Awaitable[None]],
+        ]
+    ]
+
+    web_socket_message_receiver: Optional[Future]
+    web_socket_message_processor: Future
+
+    proxy: Optional[str]
+    ping_interval: float
+    current_session: Optional[ClientWebSocketResponse]
+    current_session_monitor: Optional[Future]
+
+    on_message_listeners: List[Callable[[WSMessage], None]]
+    on_error_listeners: List[Callable[[WSMessage], None]]
+    on_close_listeners: List[Callable[[WSMessage], None]]
+
+    def __init__(
+        self,
+        app_token: str,
+        logger: Optional[Logger] = None,
+        web_client: Optional[AsyncWebClient] = None,
+        proxy: Optional[str] = None,
+        auto_reconnect_enabled: bool = True,
+        ping_interval: float = 10,
+        on_message_listeners: Optional[List[Callable[[WSMessage], None]]] = None,
+        on_error_listeners: Optional[List[Callable[[WSMessage], None]]] = None,
+        on_close_listeners: Optional[List[Callable[[WSMessage], None]]] = None,
+    ):
+        self.app_token = app_token
+        self.logger = logger or logging.getLogger(__name__)
+        self.web_client = web_client or AsyncWebClient()
+        self.proxy = proxy
+        self.auto_reconnect_enabled = auto_reconnect_enabled
+        self.ping_interval = ping_interval
+
+        self.wss_uri = None
+        self.web_socket_message_queue = Queue()
+        self.web_socket_message_listeners = []
+        self.socket_mode_request_listeners = []
+        self.current_session = None
+        self.current_session_monitor = None
+
+        self.on_message_listeners = on_message_listeners or []
+        self.on_error_listeners = on_error_listeners or []
+        self.on_close_listeners = on_close_listeners or []
+
+        self.web_socket_message_receiver = None
+        self.web_socket_message_processor = asyncio.ensure_future(
+            self.process_web_socket_messages()
+        )
+
+    async def monitor_current_session(self) -> None:
```

Comment:
This is a background job that checks if the current session is heathy. If it detects an issue, this app tries to reconnect with a new WSS URL.